scripts/update_todo_references.py

```diff
@@ -0,0 +1,248 @@
+    auth = HTTPBasicAuth(JIRA_USERNAME, JIRA_API_TOKEN)
+    headers = {"Accept": "application/json", "Content-Type": "application/json"}
+
+    payload = json.dumps(
+        {"issueUpdates": [issue_payload_for_todo(todo) for todo in todos]}
+    )
+
+    response = requests.request(
+        "POST", url, data=payload, headers=headers, auth=auth
+    )
+
+    issues = json.loads(response.text)["issues"]
+    for todo, issue in zip(todos, issues):
+        todo.issue_key = issue["key"]
+
+
+def get_all_jira_todo_issues() -> List[JiraIssue]:
+    """Returns a list of all JIRA issues in the board specified by
+    `JIRA_BOARD_ID` and the label specified in `JIRA_ISSUE_LABEL`."""
+    url = f"{JIRA_BASE_URL}/rest/api/3/search"
+    auth = HTTPBasicAuth(JIRA_USERNAME, JIRA_API_TOKEN)
+    headers = {"Accept": "application/json", "Content-Type": "application/json"}
+
+    query = {
+        "jql": f'project = {JIRA_BOARD_ID} AND labels = "{JIRA_ISSUE_LABEL}"',
+        "fields": "status",
+        "maxResults": 500,
+    }
+
+    response = requests.request(
+        "GET", url, params=query, headers=headers, auth=auth
+    )
+
+    issues = []
+    for issue_dict in response.json()["issues"]:
+        status_category = issue_dict["fields"]["status"]["statusCategory"]["id"]
+        is_done = status_category == JIRA_DONE_STATUS_CATEGORY_ID
+        issues.append(JiraIssue(key=issue_dict["key"], done=is_done))
+
+    return issues
+
+
+def remove_todos_for_closed_issues(
+    file: Path, todos: List[Todo], issues: List[JiraIssue]
+) -> None:
+    """Removes Todos if their referenced issue was closed.
+
+    Args:
+        file: The file for which to remove the todos.
+        todos: All todos with issue reference found in this file.
+        issues: All jira issues referencing code todos.
+    """
+    open_issue_keys = {issue.key for issue in issues if not issue.done}
+    todos_to_delete = [
+        todo.issue_key
+        for todo in todos
+        if todo.issue_key not in open_issue_keys
+    ]
+
+    if todos_to_delete:
+        file_content = file.read_text()
+
+        regex = re.compile(
+            r"^[ \t]*# TODO ?\[("
+            + "|".join(todos_to_delete)
+            + r")\] ?:(.*$\n(^[ \t]*# {2}.*$\n)*)",
+            flags=re.MULTILINE
+        )
+        new_file_content = re.sub(
+            regex,
+            "",
+            file_content,
+        )
+
+        file.write_text(new_file_content)
+        print(f"Deleted {len(todos_to_delete)} todos in file '{file}'.")
+
+
+def find_todos(file: Path) -> Tuple[List[Todo], List[Todo]]:
+    """Finds referenced and unreferenced Todos in a file.
+
+    Args:
+        file: The file to search.
+
+    Returns: A tuple (unreferenced_todos, referenced_todos) for todos
+        without/with reference to jira issues.
+    """
+    file_content = file.read_text()
+
+    matches = re.findall(
+        r"^[ \t]*# TODO ?\[(LOW|MEDIUM|HIGH|[A-Z]*?-[0-9]*?)\] ?:"
```

Comment:
HIGHEST and LOWEST could also be a thing

Reply:
Wasn't sure if we'd actually need them, but probably makes sense to simply include them here